flood_adapt/adapter/sfincs_adapter.py

```diff
@@ -1,210 +1,650 @@
-            ds["lon"] = ds["lon"] - 360
-        self.sf_model.setup_wind_forcing_from_grid(wind=ds)
+        if isinstance(measure, FloodWall):
+            self._add_measure_floodwall(measure)
+        elif isinstance(measure, GreenInfrastructure):
+            self._add_measure_greeninfra(measure)
+        elif isinstance(measure, Pump):
+            self._add_measure_pump(measure)
+        else:
+            self.logger.warning(
+                f"Skipping unsupported measure type {measure.__class__.__name__}"
+            )
 
-    def add_pressure_forcing_from_grid(self, ds: xr.DataArray):
-        """Add spatially varying barometric pressure to sfincs model.
+    def add_projection(self, projection: IProjection):
+        """Get forcing data currently in the sfincs model and add the projection it."""
+        self.logger.info("Adding Projection to the SFINCS model...")
+        phys_projection = projection.get_physical_projection()
 
-        Parameters
-        ----------
-        ds : xr.DataArray
-            Dataarray which should contain:
-            - press: barometric pressure [Pa]
-            - spatial_ref: CRS
-        """
-        self.sf_model.setup_pressure_forcing_from_grid(press=ds)
+        if phys_projection.attrs.sea_level_rise:
+            self.logger.info(
+                f"Adding sea level rise ({phys_projection.attrs.sea_level_rise}) to SFINCS model."
+            )
+            self.waterlevels += phys_projection.attrs.sea_level_rise.convert(
+                us.UnitTypesLength.meters
+            )
 
-    def add_precip_forcing_from_grid(self, ds: xr.DataArray):
-        """Add spatially varying precipitation to sfincs model.
+        # ? phys_projection.attrs.subsidence
 
-        Parameters
-        ----------
-        precip : xr.DataArray
-            Dataarray which should contain:
-            - precip: precipitation rates [mm/hr]
-            - spatial_ref: CRS
+        if phys_projection.attrs.rainfall_multiplier:
+            self.logger.info(
+                f"Adding rainfall multiplier ({phys_projection.attrs.rainfall_multiplier}) to SFINCS model."
+            )
+            if self.rainfall is not None:
+                self.rainfall *= phys_projection.attrs.rainfall_multiplier
+            else:
+                self.logger.warning(
+                    "Failed to add rainfall multiplier, no rainfall forcing found in the model."
+                )
+
+    ### GETTERS ###
+    def get_model_root(self) -> Path:
+        return Path(self._model.root)
+
+    def get_mask(self):
+        """Get mask with inactive cells from model."""
+        mask = self._model.grid["msk"]
+        return mask
+
+    def get_bedlevel(self):
+        """Get bed level from model."""
+        self._model.read_results()
+        zb = self._model.results["zb"]
+        return zb
+
+    def get_model_boundary(self) -> gpd.GeoDataFrame:
+        """Get bounding box from model."""
+        return self._model.region
```

Comment:
My idea with these is that
getters are static data related to the template SfincsModel in the database (will always be in the same place for hydromt-sfincs)
properties are forcings we add to the model on a per scenario basis (depending on which type of forcing we add, it has to return something else since hydromt-sfincs stores it in some other place and there is no hydromt-sfincs function to get it)

Reply:
that's a decent enough distinction I guess.